OOP_Les1/Home.py

```diff
@@ -0,0 +1,149 @@
+
+    Exchange_Rate = {'USD': {'rate': 1,
+                             'sign': '$'},
+                     'EUR': {'rate': 1.09,
+                             'sign': "\u20AC"},
+                     'UAH': {'rate': 0.027,
+                             'sign': "\u20B4"}}
+
+    def __init__(self, card_num, cvv, pin, name, surname, end_date, balance=Card_Credit_Limit):
+        """Initialize card: add card number validation as a static method."""
+        if not BankCard.validate_card_number(card_num):
+            raise ValueError('Invalid card number')
+        self.card_num = card_num
+        self.__cvv = cvv
+        self.__pin = pin
+        self.name = name
+        self.surname = surname
+        self.end_date = end_date
+        self.__balance = balance
+
+    def __str__(self):
+        """Return message about card creation."""
+        return f'Bank card for {self.name} {self.surname}'
+
+    @property
+    def card_balance(self):
+        """Return card Balance."""
+        return self.__balance
+
+    @card_balance.setter
+    def card_balance(self, balance):
+        """Update Card Balance."""
+        self.__balance = balance
+
+    @property
+    def cvv(self):
+        """Return card CVV."""
+        return self.__cvv
+
+    @cvv.setter
+    def cvv(self, new_cvv):
+        """Update CVV."""
+        self.__cvv = new_cvv
+
+    @property
+    def pin(self):
+        """Return card PIN."""
+        return self.__pin
+
+    @pin.setter
+    def pin(self, new_pin):
+        """Set card PIN."""
+        self.__pin = new_pin
+
+    @staticmethod
+    def generate_new_cvv():
+        """Generate new CVV."""
+        return secrets.randbelow(900) + 100
+
+    @staticmethod
+    def validate_card_number(card_num):
+        """Validate card number."""
+        return len(card_num) == 16 and card_num.isdigit()
+
+    @classmethod
+    def get_exchange_rate(cls, currency):
+        """Get exchange rate."""
+        return cls.Exchange_Rate[currency]['rate']
+
+    def withdraw(self, sum, currency='USD'):
+
+        """Withdraw money from card.
+        Raise ValueError if not enough money.
+        Exchange rate should be taken from class attributes.
+        """
+        sign = BankCard.Exchange_Rate[currency]['sign']
+        # Check that card not Expired
+        if self.validate_end_date():
+            raise ValueError('Card is Expired. Please request a new card.')
+
+        if currency:
+            rate = BankCard.get_exchange_rate(currency)
+            sum = (sum * rate).__round__(2)
+            if currency != 'USD':
+                print(f'Exchange rate: 1$/{rate}{sign}.')
+
+        if sum > self.card_balance:
+            raise ValueError('Not enough money')
+        self.card_balance = (self.card_balance - sum).__round__(2)
+        print(f'Withdrew {sum}$. Balance: {self.card_balance}$')
+
+    def validate_end_date(self):
+        Card_Expired = False
+        # Determine today's date
+        date = datetime.now()
```

Comment:
For date and time comparison better to use best practice 
See examples here
https://pynative.com/python-compare-dates/